cwl/src/main/scala/cwl/CommandOutputExpression.scala

```diff
@@ -0,0 +1,128 @@
+                    Map("contents" -> load64KiB(path, ioFunctionSet)).filter(_ => _loadContents)
+
+                Map(
+                  "location" -> path
+                ) ++ contents
+            }).toArray
+
+          val outputEvalParameterContext: ParameterContext = parameterContext.setSelf(womMaps)
+
+          expression.fold(EvaluateExpression).apply(outputEvalParameterContext).toEither.leftMap(e => NonEmptyList.one(e.getMessage))
+      }
+    }
+    //To facilitate ECMAScript evaluation, filenames are stored in a map under the key "location"
+    val womValue =
+      commandOutputBindingToWomValue map {
+        case WomArray(_, Seq(WomMap(WomMapType(WomStringType, WomStringType), map))) => map(WomString("location"))
+        case other => other
+      }
+
+
+
+    //If the value is a string but the output is expecting a file, we consider that string a POSIX "glob" and apply
+    //it accordingly to retrieve the file list to which it expands.
+    val globbedIfFile:ErrorOr[WomValue] =
+      (womValue, cwlExpressionType) match {
+
+        //In the case of a single file being expected, we must enforce that the glob only represents a single file
+        case (Right(WomString(glob)), WomSingleFileType) =>
+          ioFunctionSet.glob(glob) match {
+            case head :: Nil => WomString(head).validNel
+            case list => s"expecting a single File glob but instead got $list".invalidNel
+          }
+
+        case (other, _) => other.toValidated
+      }
+
+    //CWL tells us the type this output is expected to be.  Attempt to coerce the actual output into this type.
+    globbedIfFile.toTry.flatMap(cwlExpressionType.coerceRawValue).toErrorOr
+  }
+
+  /*
+  TODO:
+   DB: It doesn't make sense to me that this function returns type WomFile but accepts a type to which it coerces.
+   Wouldn't coerceTo always == WomFileType, and if not then what?
+   */
+  override def evaluateFiles(inputs: Map[String, WomValue], ioFunctionSet: IoFunctionSet, coerceTo: WomType): ErrorOr[Set[WomFile]] ={
+
+    ParameterContext().addInputs(inputs).
+      map(pc =>
+        for {
+          globValue <- outputBinding.glob.toList
+          path <- GlobEvaluator.globPaths(globValue, pc, ioFunctionSet).toList
+        } yield WomGlobFile(path): WomFile).
+      map(_.toSet).
+      toValidated
+
+  }
+
+  private def load64KiB(path: String, ioFunctionSet: IoFunctionSet): String = {
+    // This suggests the IoFunctionSet should have a length-limited read API as both CWL and WDL support this concept.
+    // ChrisL: But remember that they are different (WDL => failure, CWL => truncate)
+    val content = ioFunctionSet.readFile(path)
+
+    // TODO: propagate IO, Try, or Future or something all the way out via "commandOutputBindingtoWomValue" signature
+    // TODO: Stream only the first 64 KiB, this "read everything then ignore most of it" method is terrible
+    val initialResult = Await.result(content, 5 seconds)
```

Comment:
5 seconds very well might not be long enough considering the "terrible"ness of this implementation.

Reply:
Strong words that I now own as I last touched this line.  I will make it less terrible and probably remove the strongly worded condemnation.